### ISSUE TITLE
Allow for creation of hierarchical "relations" in taxonomy terms

### ISSUE THREAD

**Issue by mjordan** [mjordan/islandora_workbench#236]
Currently, Workbench can only create new terms that are direct children of the vocabulary's root. It would be useful to create parent-child relations (to use the term in the "Relations" section of each terms' edit form), on the fly. We need to accommodate new and existing term names and existing term IDs as parents, and squeeze this all into a CSV values that currently look like this:

`"tags:gum, Bubble|tags:candy, Hard"`

This ability should also be available in typed relation fields, which can currently look as complex as:

`"relators:art:person:Jordan, Mark|relators:art:45"`

And, of course, new terms with relations will need to be validated, i.e., their structure and whether the parent exists.

**Comment by mjordan** [comment]
All we need to do to identify where a new terms fits into the vocabulary's hierarchy is pair the new term with its intended parent. The way we pair it needs to be some character or set of characters that would not likely appear in terms, or that Workbench doesn't use for another purpose already. Here is a set of examples using `>>` to separate the new term from its intended parent, using the following simple vocabulary:

```
root
    term A (id 5)
        term B (id 20)
            term C (id 30)
```
To create a new term `new term` under `term C`, the CSV value would look like `term C >> new term` or `30 >> new term`. To create a new term `another new term` under `term B`, the CSV value would look like `term B` >> `another new term` or `20 > another new term`.

Parent term URIs would also be allowed: if `term B`'s URI is `http://example.com/foo`, a valid CSV value would be `http://example.com/foo >> another new term`.

Here is this notation used within the examples above of multivalued CSV values and taxonomy terms within typed relation CSV values:

* We want to create a new term "gum, Bubble" as a child of the term "candy" within the "tags" vocabulary
   * `"tags:candy >> gum, Bubble|tags:candy, Hard"`
* We want to create a new term "Jordan, Mark" as a child of the term "Librarians" within the "person" vocabulary
   * `"relators:art:person:Librarians >> Jordan, Mark|relators:art:45"`

Would love some feedback on this proposal.

**Comment by mjordan** [comment]
WIP on this in #111.

**Comment by mjordan** [comment]
Resolved with 0cf874651f540f227c016a99dadf50c8ce507867.
